Source/WebKit/UIProcess/XR/ios/PlatformXRARKit.h

```diff
@@ -72,12 +72,13 @@ class ARKitCoordinator final : public PlatformXRCoordinator {
         RetainPtr<id<WKARPresentationSession>> presentationSession;
         RefPtr<Thread> renderThread;
         Box<BinarySemaphore> renderSemaphore;
-    };
-    struct Terminating {
-        WeakPtr<PlatformXRCoordinator::SessionEventClient> sessionEventClient;
+        // NOTE: Since we only require atomic load/store, does this need to be
+        // atomic? Making terminateRequested atomic breaks the ability to easily
+        // construct Active inside std::variant.
+        bool terminateRequested;
```

Comment:
an std::atomic would be easier IMHO.

Reply:
It doesn't compile with `std::atomic` since that can't be moved.